### ISSUE TITLE
regression: omake 0.10.1 doesn't compile ppxx

### ISSUE THREAD

**Issue by ilitzroth** [ocaml-omake/omake#62]
$ opam switch install omakeproblem --yes --alias-of 4.03.0
$  eval `opam config env`
$ opam install ppxx

leads to:
>#=== ERROR while installing ppxx.1.3.0 ========================================#
># opam-version 1.2.2
># os           linux
># command      ocaml setup.ml -build
># path         /home/immanuel/.opam/omakeproblem/build/ppxx.1.3.0
># compiler     4.03.0
># exit-code    1
># env-file     /home/immanuel/.opam/omakeproblem/build/ppxx.1.3.0/ppxx-20715-f0f09f.env
># stdout-file  /home/immanuel/.opam/omakeproblem/build/ppxx.1.3.0/ppxx-20715-f0f09f.out
># stderr-file  /home/immanuel/.opam/omakeproblem/build/ppxx.1.3.0/ppxx-20715-f0f09f.err
>### stdout ###
># *** omake: reading OMakefiles
>### stderr ###
># [...]
># File "setup.ml", line 5848, characters 11-30:
># Warning 3: deprecated: String.uncapitalize
># Use String.uncapitalize_ascii instead.
># *** omake error:
>#    File Omake_builtin: line 0, characters 0-0
>#    include file Pervasives not found in OMAKEPATH
>#       (OMAKEPATH[] =
>#           "." : Sequence
>#           /home/immanuel/.opam/omakeproblem/lib/omake : Dir)
># E: Failure("Command 'sh -c \"cp OMakeroot.in OMakeroot; omake >PREFIX=/home/immanuel/.opam/omakeproblem\"' terminated with error code 123")

but pinning omake to version 0.9.8.6-0.rc1 before installing ppxx works without 
a hitch

**Comment by gerdstolpmann** [comment]
I cannot reproduce the error:
```
$ opam install ppxx
The following actions will be performed:
  ∗  install ppxx 1.3.1

=-=- Gathering sources =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
[default] https://opam.ocaml.org/archives/ppxx.1.3.1+opam.tar.gz downloaded

=-=- Processing actions -=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
∗  installed ppxx.1.3.1
Done.
gerd@e130:~/tmp$ omake --version
OMake 0.10.1 (development snapshot):\n\tbuild [Tue Nov 22 14:45:45 2016]\n\ton e130
```

**Comment by rgrinberg** [comment]
Cannot reproduce either on 4.04

**Comment by graymalkin** [comment]
I've also just bumped into this:

```
$ opam install ppxx
The following actions will be performed:
  - install omake 0.9.8.6-0.rc1                         [required by ppxx]
  - install ppxx  1.2.1        
===== 2 to install =====
Do you want to continue ? [Y/n] y

=-=- Gathering sources =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
[default] https://opam.ocaml.org/archives/ppxx.1.2.1+opam.tar.gz downloaded
[default] https://opam.ocaml.org/archives/omake.0.9.8.6-0.rc1+opam.tar.gz downloaded

=-=- Processing actions -=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
[ERROR] The compilation of omake failed at "make bootstrap PREFIX=/home/simon/.opam/system".

#=== ERROR while installing omake.0.9.8.6-0.rc1 ===============================#
# opam-version 1.2.2
# os           linux
# command      make bootstrap PREFIX=/home/simon/.opam/system
# path         /home/simon/.opam/system/build/omake.0.9.8.6-0.rc1
# compiler     system (4.02.3)
# exit-code    2
# env-file     /home/simon/.opam/system/build/omake.0.9.8.6-0.rc1/omake-941-ad6566.env
# stdout-file  /home/simon/.opam/system/build/omake.0.9.8.6-0.rc1/omake-941-ad6566.out
# stderr-file  /home/simon/.opam/system/build/omake.0.9.8.6-0.rc1/omake-941-ad6566.err
### stdout ###
# [...]
# cc  -I"`ocamlc -where`" -c lm_notify.c
# cc  -I"`ocamlc -where`" -c lm_termsize.c
# cc  -I"`ocamlc -where`" -c lm_terminfo.c
# cc  -I"`ocamlc -where`" -c lm_fs_case_sensitive.c
# rm -f clib.a
# ar cq clib.a lm_heap.o lm_channel.o lm_printf.o lm_ctype.o lm_uname_ext.o lm_unix_cutil.o lm_compat_win32.o readline.o omake_shell_sys.o fam_win32.o fam_kqueue.o fam_inotify.o lm_notify.o lm_termsize.o lm_terminfo.o lm_fs_case_sensitive.o
# ocamlc -g -custom -o omake -cclib clib.a unix.cma  lm.cma util.cma magic.cma ast.cma ir.cma env.cma exec.cma eval.cma shell.cma build.cma builtin.cma omake_shell.cmo omake_main.cmo
# Makefile:963: recipe for target 'omake' failed
# make[1]: Leaving directory '/home/simon/.opam/system/build/omake.0.9.8.6-0.rc1/boot'
# Makefile:21: recipe for target 'bootstrap' failed
### stderr ###
# [...]
# /usr/bin/ld: clib.a(lm_unix_cutil.o): relocation R_X86_64_32S against symbol `stderr@@GLIBC_2.2.5' can not be used when making a shared object; recompile with -fPIC
# /usr/bin/ld: clib.a(readline.o): relocation R_X86_64_32S against `.bss' can not be used when making a shared object; recompile with -fPIC
# /usr/bin/ld: clib.a(lm_termsize.o): relocation R_X86_64_32S against symbol `caml_local_roots' can not be used when making a shared object; recompile with -fPIC
# /usr/bin/ld: clib.a(lm_terminfo.o): relocation R_X86_64_32S against symbol `caml_local_roots' can not be used when making a shared object; recompile with -fPIC
# /usr/bin/ld: final link failed: Nonrepresentable section on output
# collect2: error: ld returned 1 exit status
# File "_none_", line 1:
# Error: Error while building custom runtime system
# make[1]: *** [omake] Error 2
# make: *** [bootstrap] Error 2



=-=- Error report -=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
The following actions were aborted
  - install ppxx 1.2.1
The following actions failed
  - install omake 0.9.8.6-0.rc1
No changes have been performed
```

```
$ opam --version
1.2.2
```

```
$ uname -a
Linux merkel 4.7.0-1-amd64 #1 SMP Debian 4.7.8-1 (2016-10-19) x86_64 GNU/Linux
```

I'm running a fairly up-to-date Debian Testing

**Comment by graymalkin** [comment]
This seems to do the trick for me:

```diff
diff --git a/mk/osconfig_unix.mk b/mk/osconfig_unix.mk
index 7ca7b6d..34827bc 100644
--- a/mk/osconfig_unix.mk
+++ b/mk/osconfig_unix.mk
@@ -23,7 +23,7 @@ OCAMLDEP = ocamldep$(OCAMLSUFFIX)
 # C configuration
 #
 CC = cc
-CFLAGS = -I"$(STDLIB)" -I"$(STDLIB)/caml" 
+CFLAGS = -I"$(STDLIB)" -I"$(STDLIB)/caml"  -fPIC
 AR = ar cq
 AROUT =
 EXT_OBJ = .o
diff --git a/src/clib/OMakefile b/src/clib/OMakefile
index accbc72..9e5d55c 100644
--- a/src/clib/OMakefile
+++ b/src/clib/OMakefile
@@ -30,6 +30,9 @@ EXTERNAL = ../libmojave-external
 
 # CGeneratedFiles($(CUTIL_FILES))
 
+CFLAGS += -fPIC
+export
+
 open configure/snprintf
 open configure/posix_spawn
 open configure/fs_case_sensitive
```

**Comment by ilitzroth** [comment]
1) problem is reproducible here in a fresh 4.04.0 opam switch too.
That's installing the switch and then immediately installing ppxx there.

**Comment by graymalkin** [comment]
@ilitzroth to work around for now (assuming you're on UNIX) clone this repo, apply that diff, and then install omake to your system

You'll need to tell opam that it's installed, to do that edit .opam/system/build/installed and add `omake 0.10.1`

**Comment by gerdstolpmann** [comment]
Please, when there is a new issue, create a new issue.

**Comment by gerdstolpmann** [comment]
Regarding the -fPIC issue, https://github.com/ocaml-omake/omake/pull/65 might be the solution. Please test.

**Comment by graymalkin** [comment]
@gerdstolpmann this looked like the same issue to me, rather than a new one.

I found out because I was compiling from opam using `opam install omake`. I just did some debugging.

**Comment by graymalkin** [comment]
Oh I see. My mistake, I mis-read the error log. I thought it was a make error installing omake, rather than an omake error installing ppx.

**Comment by rgrinberg** [comment]
@ilitzroth your original issue should be fixed. Please try again after doing `$ opam update` and reinstalling omake.

**Comment by ilitzroth** [comment]
Problem is now fixed over here. 
Good work!

**Comment by rgrinberg** [comment]
The original issue reported here has been fixed. The other issue mentioned by @graymalkin has a fix pending #65 

Thank you to everyone involved.
